### PR TITLE
MRG: Allow connectivity=False

### LINKED ISSUE #4959: API/ENH: allow to do pure maximum statistic permutation tests without clustering
Maximum statistic tests, like implemented in ```mne.stats.permutation_t_test``` are popular choices in M/EEG http://onlinelibrary.wiley.com/doi/10.1111/j.1469-8986.2011.01273.x/full
I feel we could easily make our clustering API perform such tests by skipping the clustering if ```connectivity=False```. This would allow us to use the same API to perform to perform general max stat tests for 1 and multiple samples with arbitrary stat functions, beyond the 1 sample t-test in ```mne.stats.permutation_t_test```. 

cc @agramfort @larsoner @SherazKhan etc.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #4959.

@dengemann can you see if this works for your purposes?

### CODE REVIEW

File: doc/python_reference.rst [hunk 1]
```diff
@@ -931,7 +931,6 @@ Non-parametric (clustering) resampling methods:
 
    permutation_cluster_test
    permutation_cluster_1samp_test
-   permutation_t_test
```
Comment (agramfort) [inline doc/python_reference.rst]: deprecated?

**Comment by agramfort** [comment]
permutation_t_test is used in tutorials and examples. If we don't document
it, it should be replaced in examples etc.
ideally it should be as fast as the new solution.

**Comment by larsoner** [comment]
Removal reverted

**Comment by larsoner** [comment]
@dengemann any time to look at this so we can get it into 0.16?

**Comment by larsoner** [comment]
@agramfort okay for you? This one seems pretty straightforward based on what we discussed in #4959

**Comment by agramfort** [comment]
update what's new?

**Comment by larsoner** [comment]
pushed

**Comment by agramfort** [comment]
thx @larsoner

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**